### PR TITLE
OpenCLSim routing using path and route

### PR DESCRIPTION
**Pull request by SiggyF**
- use routable to compute distance along a route (list of nodes) and a path (linestring)
- update to latest black format
- add some extra documentation
- update to latest pytest

### CODE REVIEW

**Comment by GerbenJdeBoer** [comment]
I added som cosmetic changes in 2 notebooks.  The notebook 32_openclsim_routing.ipynb seems not necesarry, it is a bare version of 32_openclsim_opentnsim_moving. 

Nice that pure OpenCLSim can use both a path and a graph. This removes the need to use OpenTNSim if you want to sail over routes in OpenCLSim.

**Comment by GerbenJdeBoer** [comment]
I updated notebook cosmtic, but the latest test now seem to fail on datetime durations ?

File: src/openclsim/core/movable.py [hunk 1]
```diff
@@ -187,18 +187,77 @@ def v(self):
 
 class Routable(Movable, Locatable):
     """Mixin class: Something with a route (networkx node list format)
-    route: a list of node ids (available on env.graph) or geometries (shapely.Geometry)
+    route: a list of node ids (available on env.graph)
+    path: a linestring used to sail over
     """
```
Comment (Pietervanhalem) [inline src/openclsim/core/movable.py:192]: can we do this in the numpy docstring style, just like all the other docstring?

```
    """
    Mixin class: Something with a route (networkx node list format).

    Parameters
    ----------
    route
        A list of node ids (available on env.graph)
    path
        A linestring used to sail over
    """
```

Comment (GerbenJdeBoer) [reply src/openclsim/core/movable.py:192]: I agree to adheere to  numpy docstring style

**Review by GerbenJdeBoer — DISMISSED** [review]
- numpy docstring style
- fix tests
- similar introduction paragraph for all notebooks

**Review by Pietervanhalem — APPROVED** [review]
Looks nice!

**Pull request merged**